p2p/simulations/http_test.go

```diff
@@ -517,3 +625,107 @@ func TestHTTPSnapshot(t *testing.T) {
 		x.connEvent(nodes[0].ID, nodes[1].ID, true),
 	)
 }
+
+func TestMsgFilterPassMultiple(t *testing.T) {
+	// start the server
+	network, s := testHTTPServer(t)
+	defer s.Close()
+
+	client := NewClient(s.URL)
+	events := make(chan *Event, 100)
+	filterstr := "prb:0;test:0"
+	sub, err := client.SubscribeNetworkWithMsgFilter(events, false, filterstr)
+	if err != nil {
+		t.Fatalf("error subscribing to network events: %s", err)
+	}
+	defer sub.Unsubscribe()
+	// check we got all the events
+
+	nodeCount := 30
+	startNetwork(network, client, t, nodeCount)
+	x := &expectEvents{t, events, sub}
+
+	filters := make(map[MsgFilter]struct{})
+	filters[MsgFilter{Proto: "prb", Code: 0}] = struct{}{}
+	filters[MsgFilter{Proto: "dum", Code: 1}] = struct{}{}
+	filters[MsgFilter{Proto: "test", Code: 0}] = struct{}{}
+	x.expectMsg(filters, 100)
+}
+
+func TestMsgFilterPassWildcard(t *testing.T) {
+	// start the server
+	network, s := testHTTPServer(t)
+	defer s.Close()
+
+	client := NewClient(s.URL)
+	events := make(chan *Event, 100)
+	filterstr := "prb:0,2-test:*"
```

Comment:
agreed; nevertheless, a correctly formatted filterstr doesn't mean we also get all expected messages and reject all messages to be filtered; thus, we probably need a few tests anyway.

Suggest to keep test amount as is, then note a TODO in order to add more unit tests for filterstr